plutus-core/cost-model/print-cost-model/Main.hs

```diff
@@ -0,0 +1,237 @@
+                               , printf "else %d" c
+                               ]
+     ConstBelowDiagonal c m -> [ "if x<y"
+                               , printf "then %s" $ intercalate "\n" (renderModel m)
+                               , printf "else %d" c
+                               ]
+     LinearOnDiagonal   f c -> [ "if x==y"
+                               , printf "then %s" $ renderLinearFunction f "x"
+                               , printf "else %d" c
+                               ]
+     -- ^ We're not properly indenting submodels in the above/below diagonal
+     -- cases, but at present our submodels all fit on one line (eg, constant or
+     -- linear).  It seems improbable that we'd ever have a submodel that
+     -- required more than one line because then we'd be dividing the plane up
+     -- into more than two regions.
+
+-- | Take a list of strings and print them line by line, the first with no extra
+-- spaces then the rest preceded by `width` spaces. The assumption is that we'll
+-- already have printed the first part of the first line.
+printListIndented :: Int -> [String] -> IO ()
+printListIndented width l =
+    case l of
+      [] -> pure ()
+      first:rest -> do
+          printf "%s\n" first
+          mapM_ (\s -> printf "%s%s\n" spaces s)  rest
+          where spaces = take width $ repeat ' '
+
+-- | Print a the name of a builtin (the Key below) and then a possibly
+-- multi-line representation of the model, alinged so that each line of the
+-- model has the same indentation.
+printModel :: ModelComponent -> Int -> (Key, CpuAndMemoryModel) -> IO ()
+printModel component width (name, CpuAndMemoryModel cpu mem) = do
+    let model = case component of {Cpu -> cpu; Memory -> mem}
+    printf "%-*s: " width (Key.toString name)
+    printListIndented (width+2) (renderModel model)  -- +2 to account for ": " after builtin name
+
+
+---------------- Command line processing ----------------
+
+usage :: IO a
+usage = do
+  prog <- getProgName
+  printf "Usage: %s [-c|--cpu|-m|--mem|--memory] [-d|--default] [<filename>]\n" prog
+  printf "\n"
+  printf "Print a JSON cost model file in readable form.\n"
+  printf "The variables x, y, z, etc. represent the *sizes* of the builtin's arguments.\n"
+  printf "Input is read from stdin if no file is given and --default is not specified.\n"
+  printf "\n"
+  printf "Options (later options take precedence over earlier ones):\n"
+  printf "   -c, --cpu (default):  print the CPU costing functions for each built-in function\n"
+  printf "   -m, --mem --memory:  print the memory costing functions for each built-in function\n"
+  printf "   -d, --default: look for the default cost model in %s\n" defaultCostModelPath
+  printf "    and print its contents if found\n"
+  printf "   <filename>: read and print the cost model in the given file\n"
+  exitSuccess
+
+parseArgs :: [String] -> IO (ModelComponent, Maybe String)
```

Comment:
I know it's a tiny thing, but I think we should just always use `optparse-applicative`.

Reply:
It was quicker for me just to write this by hand than to remember how to use `optparse-applicative`!

Reply:
But it's easier to maintain if it uses it. And if it's checked in then we're maintaining it.